### PR TITLE
Adds the Smoking Room icebox surface ruin

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

My contribution to the March Into Mapping/Mapness event -- The forlorn homestead of mister Charles Morlbaro, a high-brow individual with a taste for cigarettes. This ruin spawns on the surface level of Icebox.

image

A local of the ice planet, Morlbaro lived a peaceful life at home, until an _incident_ compromised one of his windows. Fearing the cold and unable to keep anything larger than a cigarette lit, he sequestered himself in his private smoking room, and hasn't been seen since.

...

Oh, right, the loot. That's the most important part for some of you guys.

image

Where there's smoke, there's fire, and where there's cigarettes, there's lighters. Mister Morlbaro collected novelty lighters, which have to be worth something to someone, right? You might be able to find some if you dig around in his belongings. Just try not to disturb him if you choose to enter his smoking room.

### Mapping March

Ckey to recieve rewards: theduffelbag

## Why It's Good For The Game

Ruin variety is nice. This one doesn't have any particularly spectacular loot, but it's a surface ruin.

Bro space ruins are sooooo 2020 bro, it's all about icebox ruins now bro.

## Changelog

:cl: Rhials
add: Adds the Smoking Room icebox ruin, found on the station level!
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
can i convince you to throw the slime zippo in there too? i think it's pretty neat. good ruin overall.

**Review by MrMelbert — CHANGES_REQUESTED** [review]
You do not appear to have the githooks setup, as your map does not have the header
`//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE` (that is to say it wasn't converted to TGM format)

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -916,6 +916,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+	icon_state = "slighter"
+	light_color = LIGHT_COLOR_ELECTRIC_CYAN
+	overlay_state = "bright"
+	grind_results = list(/datum/reagent/iron = 1, /datum/reagent/flash_powder = 10)
+	light_range = 8
+	light_power = 3 //Irritatingly bright and large enough to cover a small room.
+	fancy = FALSE
+
+/obj/item/lighter/bright/examine(mob/user)
+	..()
```
Comment (Fikou) [inline code/game/objects/items/cigs_lighters.dm:953]: you need to include return value

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -916,6 +916,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	overlay_state = "slime"
 	grind_results = list(/datum/reagent/iron = 1, /datum/reagent/fuel = 5, /datum/reagent/medicine/pyroxadone = 5)
 
+/obj/item/lighter/skull
+	name = "badass zippo"
+	desc = "An absolutely badass zippo lighter. Just look at that skull!"
+	custom_price = PAYCHECK_CREW * 6
```
Comment (ArcaneMusic) [inline code/game/objects/items/cigs_lighters.dm:922]: This doesn't give it a sale value, you'll need to make a cargo export datum for it to be worth that much in cargo.
Check the `cargo/exports` folder for examples.

**Review by ArcaneMusic — CHANGES_REQUESTED** [review]

**Comment by Loafin34** [comment]
If only I knew modsuit code, So I could make a lighter module... alas.

**Review by Maurukas — CHANGES_REQUESTED** [review]
The ruin itself looks good, however you're saving in .dmm format. You need to convert the map to tgm format. You can do this by setting up the git hooks, or by opening the map in StrongDMM2 and going to File -> Preferences and changing the Save Format to TGM (then saving).

Mark ready for review when you're done, or we can help in mapping-general on the discord.

**Review by Maurukas — APPROVED** [review]
Maps look good. However, the smoke effect on the corpse is runtiming for me.

[18:30:14] Runtime in remains.dm,42: Cannot read /datum/reagent/consumable/nutr... (/datum/reagent/consumable/nutraslop).name
  proc name: release smoke (/obj/effect/decal/remains/human/smokey/proc/release_smoke)
  usr: Maurukas/(Vire-Sartri)
  usr.loc: (Smoking Room (226,211,4))
  src: the remains (/obj/effect/decal/remains/human/smokey)
  src.loc: the carpet (226,211,4) (/turf/open/floor/carpet/blue)

I'm not immediately sure why it's doing this, I would bother a coder.

File: code/game/objects/effects/decals/remains.dm [hunk 3]
```diff
@@ -39,7 +39,7 @@
 			COOLDOWN_START(src, gas_cooldown, rand(20 SECONDS, 2 MINUTES))
 
 /obj/effect/decal/remains/human/smokey/proc/release_smoke(mob/living/smoke_releaser)
-	visible_message(span_warning("[smoke_releaser] disturbs the [src], which releases a huge cloud of [that_shit_that_killed_saddam.name]!"))
+	visible_message(span_warning("[smoke_releaser] disturbs the [src], which releases a huge cloud of gas!"))
```
Comment (LemonInTheDark) [inline code/game/objects/effects/decals/remains.dm:42]: worth noting you coulda used initial(that_shit_that_killed_sadam.name), it's how you access non list vars off typepaths

**Pull request merged**